src/app/constants.ts

```diff
@@ -0,0 +1,12 @@
+import { Theme, ThemeName } from "./types";
+
+export const themes: Record<ThemeName, Theme> = {
```

Comment:
this makes it seem like each component will look up the theme like so:
```typescript
...
<ThemeConsumer>
    {themeName => {
        const theme = themes[themeName];
        
        <div style={{background: theme.background}} />
    }}
</ThemeConsumer>
...
```

But that method requires you to import a global variable `themes` in order to do the lookup. Instead you should pass down the actual theme through context.
```typescript
...
<ThemeConsumer>
    {({ background }) => <div style={{background}} />}
</ThemeConsumer>
...
```

Or more flexibly, the context could look like this:
```typescript
type ThemeContext = {
    activeTheme: Theme;
    themes: Theme[];
};
```
That way you can render some "theme-changer" component and you'll have access to the active theme and all themes.